pandas/io/parsers.py

```diff
@@ -1387,6 +1390,20 @@ def _wrap_compressed(f, compression, encoding=None):
             data = bz2.decompress(f.read())
             f = StringIO(data)
         return f
+    elif compression == 'zip':
```

Comment:
ok then is possible to isolate this so each is just a function call to a common function? (just trying to avoid this code duplication)